modules/drivers/redshift/test/metabase/driver/redshift_test.clj

```diff
@@ -192,3 +196,29 @@
                    (map
                     (partial into {})
                     (db/select [Field :name :database_type :base_type] :table_id table-id))))))))))
+
+(defn- assert-jdbc-url-fetch-size [db fetch-size]
+  (let [ds   (sql-jdbc.execute/datasource db)
+        conn (.getConnection ds)
+        md   (.getMetaData conn)
+        url  (.getURL md)]
+    (is (str/includes? url (str "defaultRowFetchSize=" fetch-size)))))
+
+(deftest test-jdbc-fetch-size
+  (testing "Redshift JDBC fetch size is set correctly in PreparedStatement"
+    (mt/test-driver :redshift
+      (assert-jdbc-url-fetch-size (mt/db) 5000) ; the default value should always be picked up if nothing is set
+      (with-redefs [env/env (assoc env/env :mb-redshift-fetch-size "14")]
+        ; we have to create a new DB in order to pick up the change to the environment here
+        (mt/with-temp Database [db {:engine :redshift, :details (:details (mt/db))}]
+          (mt/with-db db
+            ;; make sure the JDBC URL has the defaultRowFetchSize parameter set correctly
+            (assert-jdbc-url-fetch-size db 14)
+            ;; now, actually run a query and see if the PreparedStatement has the right fetchSize set
+            (qp.store/with-store
+              (qp.store/fetch-and-store-database! (u/the-id db))
```

Comment:
you don't need to do all this yourself, just use `mt/with-everything-store` in your tests

Reply:
Changed.